### ISSUE TITLE
Server closed the connection

### ISSUE THREAD

**Issue by saeed-rz** [processone/ejabberd#1015]
> What version of ejabberd are you using?

im using ejabberd 16.02.74

> What operating system (version) are you using?

ubuntu 14.04

> How did you install ejabberd (source, package, distribution)?

source

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

after compile and install when i want to login with pidgin client to ejabberd, pidgin say 
**Server closed the connection**

and ejabberd log is :
```
2016-03-17 17:36:45.482 [info] <0.1699.0>@ejabberd_listener:accept:333 (#Port<0.59067>) Accepted connection 192.168.1.102:49470 -> 1$
2016-03-17 17:36:45.483 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<?xml version='1.0' ?><str$
2016-03-17 17:36:45.483 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<?xml version='1.0'?><stream:stream x$
2016-03-17 17:36:45.484 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<stream:features><c xmlns='http://jab$
2016-03-17 17:36:45.489 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<auth xmlns='urn:ietf:para$
2016-03-17 17:36:45.489 [debug] <0.1732.0>@shaper:update:120 State: {maxrate,1000,0.0,1458223605484475}, Size=472
M=236.0, I=5.123
2016-03-17 17:36:45.490 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<challenge xmlns='urn:ietf:params:xml$
2016-03-17 17:36:45.721 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<response xmlns='urn:ietf:$
2016-03-17 17:36:45.722 [debug] <0.1732.0>@shaper:update:120 State: {maxrate,1000,998.6754854790046,1458223605720788}, Size=386
M=385.4894136738989, I=1.276
2016-03-17 17:36:45.722 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<challenge xmlns='urn:ietf:params:xml$
2016-03-17 17:36:46.108 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<response xmlns='urn:ietf:$
2016-03-17 17:36:46.108 [debug] <0.1732.0>@shaper:update:120 State: {maxrate,1000,998.6818684603818,1458223606107295}, Size=52
M=51.93154738948474, I=1.4
2016-03-17 17:36:46.109 [info] <0.1733.0>@ejabberd_c2s:wait_for_sasl_response:883 ({socket_state,gen_tcp,#Port<0.59067>,<0.1732.0>})$
2016-03-17 17:36:46.109 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<success xmlns='urn:ietf:params:xml:n$
2016-03-17 17:36:46.161 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<stream:stream to='192.168$
2016-03-17 17:36:46.161 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<?xml version='1.0'?><stream:stream x$
2016-03-17 17:36:46.162 [debug] <0.1732.0>@shaper:update:120 State: {maxrate,1000,992.8119333191676,1458223606159983}, Size=118
M=117.15786147950162, I=1.955
2016-03-17 17:36:46.162 [debug] <0.1733.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<stream:features><c xmlns='http://jab$
2016-03-17 17:36:46.279 [debug] <0.1732.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<iq type='set' id='purplee$
2016-03-17 17:36:46.280 [debug] <0.1732.0>@shaper:update:120 State: {maxrate,1000,0.0,1458223606162293}, Size=88
M=44.0, I=117.796
2016-03-17 17:36:46.280 [error] <0.1733.0> gen_fsm <0.1733.0> in state wait_for_bind terminated with reason: bad argument in call to$
2016-03-17 17:36:46.280 [error] <0.1733.0> CRASH REPORT Process <0.1733.0> with 0 neighbours exited with reason: bad argument in cal$
```

**Comment by badlop** [comment]
Ohh, in your log there are some error lines at the end, but you didn't show them completely, maybe they give some clue about the problem. Can you get them again and post them completely?

**Comment by saeed-rz** [comment]
oh, sorry

this is full log

2016-03-17 19:13:13.751 [info] <0.1699.0>@ejabberd_listener:accept:333 (#Port<0.59114>) Accepted connection 192.168.1.102:50120 -> 192.168.1.102:5222
2016-03-17 19:13:13.755 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<?xml version='1.0' ?><stream:stream to='192.168.1.102' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' version='1.0'>">>
2016-03-17 19:13:13.756 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<?xml version='1.0'?><stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' id='3332901100031508899' from='192.168.1.102' version='1.0' xml:lang='en'>">>
2016-03-17 19:13:13.759 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<stream:features><c xmlns='http://jabber.org/protocol/caps' hash='sha-1' node='http://www.process-one.net/en/ejabberd/' ver='KEp0RAeDtSKlZuKh8Zozb38WZ8M='/><register xmlns='http://jabber.org/features/iq-register'/><mechanisms xmlns='urn:ietf:params:xml:ns:xmpp-sasl'><mechanism>PLAIN</mechanism><mechanism>DIGEST-MD5</mechanism><mechanism>X-OAUTH2</mechanism><mechanism>SCRAM-SHA-1</mechanism></mechanisms></stream:features>">>
2016-03-17 19:13:13.761 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<auth xmlns='urn:ietf:params:xml:ns:xmpp-sasl' mechanism='DIGEST-MD5' xmlns:ga='http://www.google.com/talk/protocol/auth' ga:client-uses-full-bind-result='true'>dXNlcm5hbWU9ImFkbWluIixyZWFsbT0iMTkyLjE2OC4xLjEwMiIsbm9uY2U9IjE1ODQwNTE1NzY4NzAzNjc1MzAxIixjbm9uY2U9Im1hUDVtSmRIaitNYXhGcVZZVUdLa0w4SS9aMlRGeHRJWjIwT3g5L1pHRXM9IixuYz0wMDAwMDAwMixxb3A9YXV0aCxtYXhidWY9NDA5NixkaWdlc3QtdXJpPSJ4bXBwLzE5Mi4xNjguMS4xMDIiLHJlc3BvbnNlPWIxZTFkMzI4MThjNjc4ZDAxODQyOGY1Mzk2NTEyNDg3</auth>">>
2016-03-17 19:13:13.761 [debug] <0.1795.0>@shaper:update:120 State: {maxrate,1000,0.0,1458228696612263}, Size=472
M=236.0, I=2.571
2016-03-17 19:13:13.763 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<challenge xmlns='urn:ietf:params:xml:ns:xmpp-sasl'>bm9uY2U9IjE2MjE0MTUwNDM3Mzc4ODg0NDYzIixxb3A9ImF1dGgiLGNoYXJzZXQ9dXRmLTgsYWxnb3JpdGhtPW1kNS1zZXNz</challenge>">>
2016-03-17 19:13:13.998 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<response xmlns='urn:ietf:params:xml:ns:xmpp-sasl'>dXNlcm5hbWU9ImFkbWluIixyZWFsbT0iMTkyLjE2OC4xLjEwMiIsbm9uY2U9IjE2MjE0MTUwNDM3Mzc4ODg0NDYzIixjbm9uY2U9IkhjRVJVamJ6Yjd3eFVsUzRvRHpJSGVlN3dEeHIzVTBwSWQ4U3U4ODAzVjA9IixuYz0wMDAwMDAwMSxxb3A9YXV0aCxtYXhidWY9NDA5NixkaWdlc3QtdXJpPSJ4bXBwLzE5Mi4xNjguMS4xMDIiLHJlc3BvbnNlPWQ4NjZhNjMwNWViNGYxZTE4MWRjMDVmMWIyYmIzMGVlLGNoYXJzZXQ9dXRmLTg=</response>">>
2016-03-17 19:13:13.998 [debug] <0.1795.0>@shaper:update:120 State: {maxrate,1000,996.7563184214083,1458228696849031}, Size=386
M=384.7519870672235, I=5.581
2016-03-17 19:13:13.999 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<challenge xmlns='urn:ietf:params:xml:ns:xmpp-sasl'>cnNwYXV0aD04ZTJhZDdiMDY5MTFkNDgxOTE0YjliNGVjZDRhNDU0OQ==</challenge>">>
2016-03-17 19:13:14.382 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<response xmlns='urn:ietf:params:xml:ns:xmpp-sasl'/>">>
2016-03-17 19:13:14.382 [debug] <0.1795.0>@shaper:update:120 State: {maxrate,1000,998.4960626542621,1458228697234940}, Size=52
M=51.921912696433694, I=7.214
2016-03-17 19:13:14.382 [info] <0.1796.0>@ejabberd_c2s:wait_for_sasl_response:883 ({socket_state,gen_tcp,#Port<0.59114>,<0.1795.0>}) Accepted authentication for admin by ejabberd_auth_internal from 192.168.1.102
2016-03-17 19:13:14.386 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<success xmlns='urn:ietf:params:xml:ns:xmpp-sasl'/>">>
2016-03-17 19:13:14.429 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<stream:stream to='192.168.1.102' xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' version='1.0'>">>
2016-03-17 19:13:14.430 [debug] <0.1795.0>@shaper:update:120 State: {maxrate,1000,995.0716706726439,1458228697287378}, Size=118
M=117.42130911861419, I=3.252
2016-03-17 19:13:14.430 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<?xml version='1.0'?><stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' id='10614538671817466445' from='192.168.1.102' version='1.0' xml:lang='en'>">>
2016-03-17 19:13:14.434 [debug] <0.1796.0>@ejabberd_c2s:send_text:1838 Send XML on stream = <<"<stream:features><c xmlns='http://jabber.org/protocol/caps' hash='sha-1' node='http://www.process-one.net/en/ejabberd/' ver='KEp0RAeDtSKlZuKh8Zozb38WZ8M='/><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'/><session xmlns='urn:ietf:params:xml:ns:xmpp-session'><optional/></session><sm xmlns='urn:xmpp:sm:2'/><sm xmlns='urn:xmpp:sm:3'/><csi xmlns='urn:xmpp:csi:0'/></stream:features>">>
2016-03-17 19:13:14.549 [debug] <0.1795.0>@ejabberd_receiver:process_data:284 Received XML on stream = <<"<iq type='set' id='purple91ae7fc3'><bind xmlns='urn:ietf:params:xml:ns:xmpp-bind'/></iq>">>
2016-03-17 19:13:14.550 [debug] <0.1795.0>@shaper:update:120 State: {maxrate,1000,0.0,1458228697292219}, Size=88
M=44.0, I=119.313
2016-03-17 19:13:14.554 [error] <0.1796.0> gen_fsm <0.1796.0> in state wait_for_bind terminated with reason: bad argument in call to p1_time_compat:unique_integer(positive) line 208 in ejabberd_c2s:new_uniq_id/0 line 1917
2016-03-17 19:13:14.554 [error] <0.1796.0> CRASH REPORT Process <0.1796.0> with 0 neighbours exited with reason: bad argument in call to p1_time_compat:unique_integer(positive) line 208 in ejabberd_c2s:new_uniq_id/0 line 1917

**Comment by prefiks** [comment]
Thank you for your report, this should be fixed by change from commit ef02053a9ddf057a1c76da3409f2218c1d778b13
